src/git-repository.coffee

```diff
@@ -111,7 +116,14 @@ class GitRepository
       @repo.release()
       @repo = null
 
-    @subscriptions.dispose()
+    if @subscriptions?
+      @subscriptions.dispose()
+      @subscriptions = null
+
+  # Public: Invoke the given callback when this GitRepository's destroy() method
+  # is invoked.
+  onDestroy: (callback) ->
```

Comment:
We always include an auxiliary verb in event subscription methods for consistency, `onDidDestroy`. Maybe this was overkill and should have only been applied to events that had a `onWill` and `onDid` case, but the :boat: has sailed.